### ISSUE TITLE
[1.25.0] Button click not working on cloud but same button is working locally

### ISSUE THREAD

**Issue by furqandaud** [mobile-dev-inc/maestro#1071]
**Describe the bug**

I am facing issue on maestro cloud run.
My tests are failing both Android and IOS because it is not clicking a button on cloud test run. When I run the test locally it is pressing the button and is being login to account but on cloud run, it fails to click button hence failing the test.

I tried using both ID and Text, still fails. Kindly help as this is first step of our app which is login.

**To Reproduce**

1. Start the simulator
2. Run maestro test`
3. Click Login button, The flow fails at clicking login button. 

**Expected behavior**

It should click login button and it should show wrong username or password error.

**Screenshots**

https://github.com/mobile-dev-inc/maestro/assets/111492691/408aa7e5-bb5d-474a-ad8c-b85702a269d5

![Screenshot 2023-05-11 at 2 49 19 PM](https://github.com/mobile-dev-inc/maestro/assets/111492691/08279c92-6676-4bec-bd9e-c9a461012e27)

**Environment information (please complete the following information):**

- Maestro version [1.25.0]
- Platform: [iOS 16.1, Android 11 Red Velvet Cake (Android API 30 )]
- Framework: [React Native]
- Device model and OS version: [iPhone 13 (iOS 16.1), Pixel 4 ( Android 11)]
- Simulator and emulator.
- Host [Mac M1 Pro, Ventura 13.2.1]

**Workspace (if applicable)**

Inside maestro folder we are running all the flows inside tests folder. The reusable flows folder is being used for using all the subflows which we are using inside tests folder to run the test. 

<img width="384" alt="Screenshot 2023-05-14 at 8 31 23 PM" src="https://github.com/mobile-dev-inc/maestro/assets/111492691/c3a8725e-c8bf-42e3-8065-c54a70d13464">

**Bugreport output (if applicable)**

I have attached hierarchy and logs in slack thread.

Link for maestro cloud console is below:
https://console.mobile.dev/uploads/1ea0164c-9ff7-4040-b22c-5ff5f6879c65?teamId=44956578-e848-4f4b-8555-fb1a75256f45&appId=1766cab8-d6bf-4113-a1a0-2156eb7fcce0&analysisId=678a18d1-00df-4dc5-bd55-69e2db22ad20&checkIdx=0

**Additional context**

https://mobile-dev-inc.slack.com/archives/C041FU72T54/p1683798900687829

Reach me out on slack anytime on above mentioned thread for any kind of information regarding this issue. 
To provide official Wahed app and whole environment, the legal team of Wahed is already in talks with Maestro so hopefully I can provide you that soon but till then I can provide all details possible when asked. 

Thanks.

**Comment by furqandaud** [comment]
> Hi Furqan, i noticed that you use Maestro 1.25.0 but the latest version is 1.27.0. Maestro also runs the latest version, 1.27.0. Can you update to the latest version locally to verify if the bug exists - that way we can determine if it's a Maestro Cloud or a Maestro CLI issue.

Hi Arthur, Okay let me update and check if the issue exists locally. I'll update you shortly.

**Comment by furqandaud** [comment]
> Hi Furqan, i noticed that you use Maestro 1.25.0 but the latest version is 1.27.0. Maestro also runs the latest version, 1.27.0. Can you update to the latest version locally to verify if the bug exists - that way we can determine if it's a Maestro Cloud or a Maestro CLI issue.

Hi Arthur, I just checked locally with the updates version. It is working fine locally.

**Comment by ArthurSav** [comment]
@furqandaud Can you try using an element id instead of text?

**Comment by furqandaud** [comment]
> @furqandaud Can you try using an element id instead of text?

Already tried that. Originally I was using element id so I thought maybe it's because of that. Only then I went onto using texy instead of element id thinking it would resolve this issue but it didn't. I can provide you the log and screenshots of the test where I am using element id.

**Comment by ArthurSav** [comment]
Ok we'll assign an engineer to look at this when possible. 
There might be a workaround till we fix the issue. You could try tapping on relative coordinates ([Cloud device screen specs](https://cloud.mobile.dev/reference/device-configuration))

**Comment by furqandaud** [comment]
> Ok we'll assign an engineer to look at this when possible. 
> There might be a workaround till we fix the issue. You could try tapping on relative coordinates ([Cloud device screen specs](https://cloud.mobile.dev/reference/device-configuration))

Thank you for your quick response. Will be looking forward to it's fix! 🙌🏻

**Comment by furqandaud** [comment]
> Ok we'll assign an engineer to look at this when possible. There might be a workaround till we fix the issue. You could try tapping on relative coordinates ([Cloud device screen specs](https://cloud.mobile.dev/reference/device-configuration))

Hi @ArthurSav I just checked everything with update and locally. Issue is coming locally too now. Not sure why it didn't come yesterday but now I have run tests both on android and ios. So I thought of clicking Login button twice and that way it works, before login I am doing hideKeyboard and also assertVisible to wait for that button so I am not sure if this was the issue. Clicking button once is not working.

**Comment by furqandaud** [comment]
> Ok we'll assign an engineer to look at this when possible. There might be a workaround till we fix the issue. You could try tapping on relative coordinates ([Cloud device screen specs](https://cloud.mobile.dev/reference/device-configuration))

I am assuming it has something to do with keyboard or hide keyboard because weird behaviour is after that.
As even on maestro cloud the keyboard and password is not visible but when I hover over that area, the keyboard boxes are visible. 

<img width="1728" alt="Screenshot 2023-05-16 at 2 32 30 PM" src="https://github.com/mobile-dev-inc/maestro/assets/111492691/8b58fdeb-d7cd-4de4-9049-a49b15c1d2ef">

**Comment by igorsmotto** [comment]
Hey @furqandaud!

I've managed to reproduce the error on **my local machine**. It seems like the behavior of your App is to require _two_ taps "Login" on the first attempt. 
For a normal user, it looks like they are first hiding the keyboard. So they naturally retry it.
I've made a screen recording without using Maestro that should clarify what I mean, notice the mouse clicks:

https://github.com/mobile-dev-inc/maestro/assets/48068670/e811adae-24a5-44a6-8991-b7caa83be0a1



A workaround is to make Maestro run the command `- tapOn: Login` two times, but I suggest you take a look if this is not a real bug in your App. To help you debug, I can say that one difference from your setup is that I'm running on a Mac Pro Intel, not sure if this is the underlying reason.
```
Maestro version [1.27.0]
Platform: [Android 11 Red Velvet Cake (Android API 30 )]
Device model and OS version: [Pixel 4 ( Android 11)]
Host [Mac Pro Intel, Ventura 13.3.1]
```
